src/app/recents/recents.component.spec.ts

```diff
@@ -0,0 +1,94 @@
+import { NodeTypeDisplayPipe, NodeExternalUrlPipe, TruncatePipe } from '../pipes/nodedisplay';
+import { ArtifactsTableComponent } from '../tables/artifacts-table/artifacts-table.component';
+import { EstuaryTableComponent } from '../tables/table.component';
+import { PropertyDisplayPipe, PropertyValueDisplayPipe } from '../pipes/propertydisplay';
+
+
+fdescribe('RecentsComponent', () => {
+  let fixture: ComponentFixture<RecentsComponent>;
+  let recentsService: RecentsService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      declarations: [
+        ArtifactsTableComponent,
+        EstuaryTableComponent,
+        NodeTypeDisplayPipe,
+        NodeExternalUrlPipe,
+        PropertyDisplayPipe,
+        PropertyValueDisplayPipe,
+        RecentsComponent,
+        TruncatePipe
+      ],
+      providers: [
+        RecentsService,
+        DatePipe
+      ],
+      imports: [
+        TabsModule.forRoot(),
+        HttpClientTestingModule,
+        NoopAnimationsModule,
+        BsDropdownModule.forRoot()
+      ],
+      // This is used to avoid having to import components we don't test here
+      schemas: [NO_ERRORS_SCHEMA]
+    }).compileComponents();
+    recentsService = TestBed.get(RecentsService);
+  });
+
+  it('should retrieve the 5 most recent advisories', fakeAsync(() => {
+    spyOn(recentsService, 'getRecents').and.returnValue(
+      of(recents)
+    );
+    fixture = TestBed.createComponent(RecentsComponent);
+    fixture.detectChanges();
+    const component = fixture.componentInstance;
+    expect(component.recentsAll).toBe(<any> recents.data);
+    const tabs = fixture.debugElement.queryAll(By.css('tab'));
+    expect(tabs.length).toBe(5);
+
+    // Ensure the tab headings are correct
+    expect(tabs[0].nativeElement.getAttribute('ng-reflect-heading')).toBe('Advisory');
+    expect(tabs[1].nativeElement.getAttribute('ng-reflect-heading')).toBe('Commit');
+    expect(tabs[2].nativeElement.getAttribute('ng-reflect-heading')).toBe('Freshmaker Event');
+    expect(tabs[3].nativeElement.getAttribute('ng-reflect-heading')).toBe('Build');
+    expect(tabs[4].nativeElement.getAttribute('ng-reflect-heading')).toBe('Bugzilla Bug');
+
+    // Ensure that table headers are correct
+    let tableHeaders = fixture.debugElement.queryAll(By.css('th'));
+    expect(tableHeaders[0].nativeElement.textContent.trim()).toBe('Advisory Name');
+    expect(tableHeaders[1].nativeElement.textContent.trim()).toBe('ID');
+    expect(tableHeaders[2].nativeElement.textContent.trim()).toBe('Security Impact');
+    expect(tableHeaders[3].nativeElement.textContent.trim()).toBe('State');
+    expect(tableHeaders[4].nativeElement.textContent.trim()).toBe('Synopsis');
+    expect(tableHeaders[5].nativeElement.textContent.trim()).toBe('Update Date');
+
+    // Ensure the correct content is being displayed after clicking a new tab
+    const tabLinks = fixture.debugElement.queryAll(By.css('.nav-link'));
+    tabLinks[1].nativeElement.click();
```

Comment:
Optional: it'd be nice to assert that the text is correct from the tab link you're clicking on. It might make debugging easier if we change the order for some reason. Alternatively, you could loop through `tabLinks` to find the tab with the expected text to avoid relying on a specific index.